pennylane_lightning/core/_serialize.py

```diff
@@ -254,23 +257,31 @@ def _pauli_sentence(self, observable, wires_map: dict = None):
         terms = [self._pauli_word(pw, wires_map) for pw in pwords]
         coeffs = np.array(coeffs).astype(self.rtype)
 
+        # TODO: Add this
+        # if len(terms) == 1 and coeffs[0] == 1.0:
+        #     return terms[0]
+
         if self.split_obs:
             return [self.hamiltonian_obs([c], [t]) for (c, t) in zip(coeffs, terms)]
         return self.hamiltonian_obs(coeffs, terms)
 
-    # pylint: disable=protected-access
+    # pylint: disable=protected-access, too-many-return-statements
     def _ob(self, observable, wires_map: dict = None):
         """Serialize a :class:`pennylane.operation.Observable` into an Observable."""
-        if isinstance(observable, Tensor):
+        if isinstance(observable, (Prod, Sum, SProd)) and observable.pauli_rep is not None:
+            return self._pauli_sentence(observable.pauli_rep, wires_map)
+        if isinstance(observable, (Tensor, Prod)):
             return self._tensor_ob(observable, wires_map)
-        if observable.name == "Hamiltonian":
+        if observable.name in ("Hamiltonian", "LinearCombination"):
             return self._hamiltonian(observable, wires_map)
         if observable.name == "SparseHamiltonian":
             return self._sparse_hamiltonian(observable, wires_map)
         if isinstance(observable, (PauliX, PauliY, PauliZ, Identity, Hadamard)):
             return self._named_obs(observable, wires_map)
-        if observable._pauli_rep is not None:
-            return self._pauli_sentence(observable._pauli_rep, wires_map)
+        if observable.pauli_rep is not None:
+            return self._pauli_sentence(observable.pauli_rep, wires_map)
+        # if isinstance(observable, (Prod, Sum)):
+        #     return self._hamiltonian(observable, wires_map)
         return self._hermitian_ob(observable, wires_map)
```

Comment:
what about raising a `QuantumFunctionError` error as the default behaviour for this method and conditionally patch to `_hermitian_ob`? :thinking: 

Something like,
```suggestion
       if isinstance(observable, (Hermitian, Projector, any_other_obs))
            return self._hermitian_ob(observable, wires_map)
       raise qml.QuantumFunctionError(f"cannot convert observable {observable}")
```

Reply:
Hmm. I personally think it's best not to raise an error as `_hermitian_ob` is a fallback for many cases (such as `Projector` as we saw earlier)

Reply:
Sounds god, @mudit2812.

Reply:
I don't have any problem with it as long as this is clearly noted in the serialization docs that Hermitian is our default fallback. I generally don't like these "default" fallback in the codebase as it adds more obscurity to the code. As discussed offline, I'd prefer a tuple for observables that we convert to Hermitian in the serialization module.